app/src/main/java/org/mozilla/fenix/settings/biometric/BiometricPromptPreferenceFragment.kt

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.biometric
+
+import android.app.Activity.RESULT_OK
+import android.app.KeyguardManager
+import android.content.Context
+import android.content.Intent
+import android.view.View
+import androidx.core.content.getSystemService
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import mozilla.components.support.base.feature.ViewBoundFeatureWrapper
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.settings.requirePreference
+
+/**
+ * Helper for creating and implementing the [BiometricPromptFeature]. Currently used
+ * for logins and credit cards.
+ */
+abstract class BiometricPromptPreferenceFragment : PreferenceFragmentCompat() {
+
+    internal val biometricPromptFeature = ViewBoundFeatureWrapper<BiometricPromptFeature>()
+
+    /**
+     * Gets the string to be used for [BiometricPromptFeature.requestAuthentication] prompting to
+     * unlock the device.
+     */
+    abstract fun unlockMessage(): String
+
+    /**
+     * Navigate when authentication is successful.
+     */
+    abstract fun navigateOnSuccess()
+
+    /**
+     * Show a warning to set up a pin/password when the device is not secured. This is only used
+     * when BiometricPrompt is unavailable on the device.
+     */
+    abstract fun showPinDialogWarning(context: Context)
+
+    /**
+     * There is a bug where while the biometric prompt is showing, you were able to quickly navigate
```

Comment:
I think this comment should be at the caller which I think is at `onPause` cause it wouldn't make sense reading this for `onAuthFailure`.

I would add a proper description here just describing the functionality of this function.

Reply:
This is a copied comment from logins. Should I update both comments or just this one?